ankerctl.py

```diff
@@ -439,13 +443,15 @@ def config_show(env):
         print()
 
         log.info("Printers:")
-        for p in cfg.printers:
+        for i, p in enumerate(cfg.printers):
```

Comment:
There's a slight problem here - because we are just iterating over `cfg.printers`, we need to make sure the ordering is consistent, to have any meaningful correlation between the index, and the intended printer.

In Python 3, the default dict type is now (if I remember correctly), always using ordered keys. So there will be some kind of correlation, but it's mostly luck.

To really make this work, we need to sort the printer list in a predictable way. For example, by serial, or `duid`.

Otherwise, we might get a new random order, every time the printer list is imported from the API, and this can happen automatically (e.g., when upgrading ankerctl).

It would be quite bad to have your printer indices randomly shuffled in this event, so we should avoid it.

Do you want some tips on how to do this, or would you like to give it a go? :)

Reply:
I added in functionality to sort by printer.id here, this is the id assigned by AnkerMake. Yes, this sort order could change if a printer is ingested with a lower id, but so can any other field that we choose to sort by. I think sorting by printer id makes the most sense as it is a constant that should never change and likely will only ever increase rather than decrease, thus the risk of the sort order getting messed up is likely reduced.

Reply:
That makes sense - and I agree, there's no 100% solution for this, that's simply impossible.

We could consider using the `create_time`, since that reflects a timestamp related to printer registration (not sure if it's when the printer was first registered, or most recently, but it should be one of those).

I'm also perfectly happy with station_id - as long as we chose something predictable, like you say.

Comments?

Reply:
> We could consider using the `create_time`, since that reflects a timestamp related to printer registration (not sure if it's when the printer was first registered, or most recently, but it should be one of those).

I tested this by de-registering my printer and re-registering it and it updated the `create_time`, so it appears to be "registration time" but is related to the most recent registration, not the first registration. 



> I'm also perfectly happy with station_id - as long as we chose something predictable, like you say.

I think `station_id` should be fairly static, during my de-registration/re-registration testing, it didnt change while `create_time` did. For that reason, I think it makes perfect sense to use this as the sorting key.

Reply:
✅